### DISCUSSION TITLE
Send sync'd Note Ons?

### DISCUSSION THREAD

**Discussion by papertree0** [mido/mido#573] [Q&A]
I'm trying to write a drum sequencer with two voices (for simplicity's sake), and I'm running into the problem that I can't send the MIDI messages to be synchronised so I'm always getting flams. 

Any idea how I would do this? Can I even do this? Is there an external library that can aid me with this? I have been looking everywhere but can't seem to find it. 

I am using a virtual port, if that's of any use here.

**Comment by rdoursenaud** [comment]
Hi,

MIDI is a serial protocol and, as such, has no way to send 2 messages perfectly in sync, these are always slightly separated when sent back to back. Exactly 320 us per byte with traditional 5-pin MIDI. So it requires a bit less than 2 ms to send two note-on messages (3 bytes each assuming running status is not used which mido currently doesn't support) with less than 1 ms delay between them.

That being said, it should not be noticeable as a flam as it is below the ~5 ms human hearing threshold.

Maybe your loop is too busy and does not send each messages back-to-back. Or maybe the engine you use to produce the sound has a bug.

I'd need more information for a proper diagnostic: code snippet, setup details…
